src/libraries/System.Private.CoreLib/generators/EventSourceGenerator.Emitter.cs

```diff
@@ -32,7 +32,7 @@ public void Emit(EventSourceClass[] eventSources, CancellationToken cancellation
                     _builder.AppendLine("using System;");
                     GenType(ec);
 
-                    _context.AddSource($"{ec.ClassName}.Generated", SourceText.From(_builder.ToString(), Encoding.UTF8));
+                    _context.AddSource($"{ec.ClassName}.g.cs", SourceText.From(_builder.ToString(), Encoding.UTF8));
```

Comment:
Why wasn't it "Generated.cs" before if the file extension is required?

Reply:
it's not required, at least not from a source gen perspective. it's convention.

Reply:
the file on disk is only for debugging purposes, so I assumed it simply didn't have .cs before. but it seems I'm wrong:
 `...generated\System.Private.CoreLib.Generators\Generators.EventSourceGenerator\RuntimeEventSource.Generated.cs`

Reply:
I'm guessing with this change it's `.g.cs.cs` ?

Reply:
> I'm guessing with this change it's .g.cs.cs ?

no, roslyn appends ".cs" if you don't include it yourself

Reply:
https://sourceroslyn.io/#Microsoft.CodeAnalysis/SourceGeneration/AdditionalSourcesCollection.cs,129